Project_Mars/Project_Mars/Pages/LanguageTab.cs

```diff
@@ -0,0 +1,270 @@
+
+            int totalrows = driver.FindElements(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table//tbody")).Count;
+            Console.WriteLine(totalrows);
+
+            if (totalrows < 4)
+            {
+                Thread.Sleep(3000);
+
+                IWebElement AddNewButton = driver.FindElement(By.XPath("//div[@data-tab='first']//div[@class ='ui teal button ']"));
+                AddNewButton.Click();
+
+                Wait.WaitToBeVisible(driver, "XPath", "//input[@placeholder='Add Language']", 3);
+                IWebElement addLanguageInputBox = driver.FindElement(By.XPath("//input[@placeholder='Add Language']"));
+                addLanguageInputBox.Click();
+                addLanguageInputBox.Clear();
+                addLanguageInputBox.SendKeys(LanguageName);
+                IWebElement selectLanguageLevelDropdown = driver.FindElement(By.XPath("//select [@class='ui dropdown']"));
+                selectLanguageLevelDropdown.Click();
+
+                if (LanguageLevel.Equals("Basic"))
+                {
+                    IWebElement level = driver.FindElement(By.XPath("//Option[@value='Basic']"));
+                    level.Click();
+                }
+                else if (LanguageLevel.Equals("Conversational"))
+                {
+                    IWebElement level = driver.FindElement(By.XPath("//Option[@value='Conversational']"));
+                    level.Click();
+                }
+                else if (LanguageLevel.Equals("Fluent"))
+                {
+                    IWebElement level = driver.FindElement(By.XPath("//Option[@value='Fluent']"));
+                    level.Click();
+                }
+                else if (LanguageLevel.Equals("Native/Bilingual"))
+                {
+                    IWebElement level = driver.FindElement(By.XPath("//Option[@value='Native/Bilingual']"));
+                    level.Click();
+                }
+                IWebElement addLanguageButton = driver.FindElement(By.XPath("//input [@type='button'][@value='Add']"));
+                addLanguageButton.Click();
+                Thread.Sleep(2000);
+            }
+
+        }
+
+
+
+        public void AssertAddedLanguageRecord(IWebDriver driver, String LanguageName)
+        {
+            driver.Navigate().Refresh();
+            Thread.Sleep(2000);
+            Wait.WaitToBeVisible(driver, "XPath", "//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[1] ", 3);
+            IWebElement AddedLanguageRecord = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[1]"));
+            Assert.That(AddedLanguageRecord.Text == LanguageName, "Language record has not added successfully");
```

Comment:
can you please move your asserts to the steps or into helper classes

Reply:
please do this everywhere

Reply:
Assertions are moved to step definition 
![image](https://github.com/Shilpii01/Project_Mars/assets/145233624/2023e12b-4b53-40a8-93df-92fa3498fa8d)